crates/oxc_sourcemap/src/encode.rs

```diff
@@ -255,6 +318,14 @@ fn test_encode() {
     }
 }
 
+#[inline]
+fn serde_escape_content<S: AsRef<str>>(s: S) -> Result<String> {
+    let mut escaped_buf = Vec::with_capacity(s.as_ref().len() + 2);
```

Comment:
Capacity should be larger. Assuming the content does contain escapes, the output will definitely be longer than the input + 2 quotes.

Question is how much extra capacity to allocate. I think it's fairly harmless to allocate *too much* capacity, as this string is just temporary and will be dropped almost immediately, whereas allocating too little capacity will cause the `Vec` to grow which involves reallocation and memory copy. So probably a generous over-estimate like `s.as_ref().len() * 2 + 2` would be OK.